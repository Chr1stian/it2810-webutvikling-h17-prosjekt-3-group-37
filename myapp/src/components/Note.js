import React, { Component } from 'react';
import './../style/note.css';
import {Card, CardActions, CardHeader, CardMedia, CardTitle, CardText} from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton'

export default class Note extends Component {

	constructor(props){
		super(props)



	}

	deleteNote = () => {
		this.props.deleteNote(this.props.note)

	}

	editNote = () => {
		this.props.editNote(this.props.note)
		
	}


	render() {
		const {note} = this.props
		return (
			<div>
				<Card className="note">
					<div>
						<div className="noteText">
						<CardHeader classname ="cardHeader" title={note.title} />
						<CardText className="cardText">{ note.text }</CardText>
						</div>
						<CardActions className="noteButtons">
							<FlatButton onClick = {this.editNote}>done</FlatButton>
							<FlatButton onClick = {this.deleteNote}>delete</FlatButton>
						</CardActions>
					</div>
				</Card>
			</div>
			)

	}


}