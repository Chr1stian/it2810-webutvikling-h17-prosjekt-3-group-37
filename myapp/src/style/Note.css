/*
Area for component. Using flex to center both vertically and horizontally
*/
.noteContainer{
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
}

/*
this contains the textfield and button. This box is centered.
*/
.textArea{
  width: 40%;
  height: 500px;

}

/*
This element centers title, inputbox and button 
*/
.textArea > div {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}
