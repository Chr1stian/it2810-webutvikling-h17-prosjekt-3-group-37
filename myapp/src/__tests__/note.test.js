import React from "react";
//Import Enzyme with shallow to test component as a unit
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
Enzyme.configure({ adapter: new Adapter() });
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
//Importing component that will be tested
import Note from "./../components/Note.js";
import "./../test/mock-localstorage.js";


test("Component renders as expected", () => {
  const component = shallow(<Note />);
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
