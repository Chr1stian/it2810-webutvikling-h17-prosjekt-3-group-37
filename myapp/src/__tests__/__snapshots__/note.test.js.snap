// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders as expected 1`] = `
<div
  className="noteContainer"
>
  <Card
    className="textArea"
    expandable={false}
    expanded={null}
    initiallyExpanded={false}
  >
    <CardTitle
      style={
        Object {
          "textAlign": "center",
        }
      }
      title="Your notes"
    />
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      fullWidth={false}
      id="textarena"
      multiLine={true}
      onChange={[Function]}
      placeholder="Do you have anything on your mind?"
      rows={13}
      rowsMax={13}
      spellCheck={false}
      style={
        Object {
          "width": "30vw",
        }
      }
      type="text"
      underlineShow={true}
      value={null}
    />
    <FlatButton
      disabled={false}
      fullWidth={false}
      labelPosition="after"
      labelStyle={Object {}}
      onClick={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
    >
      Delete
    </FlatButton>
  </Card>
</div>
`;
